### PR TITLE
Signature port

### LINKED ISSUE #47: Cache not being used as cache
`static PlayerTable** player_table_ptr = reinterpret_cast<PlayerTable**>(sig_player_table.address());`
In some places you use static to cache values, but then reassign to that value every time, thus negating any benefits you'd get from caching.

### LINKED ISSUE #48: Rework hooker
There is a lot of bad things going on in there that need to be addressed.

It was written when I only barely understood the parts of the language that I was using.

### LINKED ISSUE #62: Move all signatures into one file
I dreaded this, but this seems to be the best idea for keeping our source files less messy.

### PR DESCRIPTION
**Pull request by ShellyHerself**
Closes #62 
Closes #47 
Closes #48

~~Needs #63 addressed before this can be dealt with. I still want to rename LiteSignature to CodeSignature for clarity.~~

### CODE REVIEW

File: vulpes/memory/global.cpp [hunk 1]
```diff
@@ -4,42 +4,30 @@
...
-char ui_name[] = "ui";
-
-char* map_name() {
-    char* name_pointer = NULL;
-    // TODO: We'll deal with this disgrace when we move all signatures into one file.
-    if (*reinterpret_cast<bool*>(sig_at_main_menu.address())){
+const char* map_name() {
+    const char* name_pointer = NULL;
+    if (at_main_menu()){
         name_pointer = &ui_name[0];
```
Comment (Mimickal) [inline vulpes/memory/global.cpp:21]: Just writing `ui_name` without any brackets or symbols should achieve this same effect. Alternately, defining it as `static const char ui_name* = "ui";` should (re: may) work.

File: hooker/hooker.hpp [hunk 2]
```diff
@@ -70,27 +57,24 @@ class CodePatch {
...
-              size_t p_size, PatchTypes p_type, T content) CPTEMPLINIT1;
-    template<typename T>
-    CodePatch(const char* d_name, intptr_t p_address,
-              size_t p_size, PatchTypes p_type, T content) CPTEMPLINIT2;
+    CodePatch(const char* d_name, uintptr_t p_address,
+              size_t p_size, PatchTypes p_type, T content) {
+        name = d_name; patch_address = p_address;
+        patch_size = p_size; type = p_type;
+        setup_internal(reinterpret_cast<void*>(&content), sizeof(T));
+    }
```
Comment (Mimickal) [inline hooker/hooker.hpp]: I'm not normally a stickler for style, but having an arbitrary number of parameters and operations per line here makes things harder to follow. Remember one concept per line, and vertical space is free.

File: vulpes/fixes/host_refusal.cpp [hunk 3]
```diff
@@ -6,25 +6,22 @@
...
-    {0x8B, 0x7F, 0x08, 0x3B, 0xFD, 0x74, 0x5F, 0x8D, 0x9B, 0x00, 0x00, 0x00, 0x00});
-Signature(false, sig_client_refusal,
-    {-1, -1, 0x75, 0x11, 0x6A, 0x01, 0x6A, 0x7E, 0x68, 0xA0, 0x00, 0x00, 0x00});
+#include "host_refusal.hpp"
 
-Patch(host_refusal_fix1, sig_host_refusal1, 0xA, 5, NOP_PATCH, 0);
-Patch(host_refusal_fix2, sig_host_refusal2, 5, {0xEB});
-Patch(client_refusal_fix, sig_client_refusal, 0, {0xEB, 0x13});
+static Patch(host_refusal_fix1, NULL, 5, NOP_PATCH, 0);
+static Patch(host_refusal_fix2, NULL, {0xEB});
```
Comment (Mimickal) [inline vulpes/fixes/host_refusal.cpp]: You have constants defined for these `{0xEB}` style values, but not here.

File: vulpes/fixes/shdr_trans_zfighting.cpp [hunk 4]
```diff
@@ -36,27 +30,36 @@ Signature(false, sig_shader_trans_zfighting3,
...
-        patch_shader_trans_zfighting3c_fix.apply();
-        patch_shader_trans_zfighting3d_fix.apply();
-    };
+    }
+    auto sig_addr3 = sig_fix_shader_trans_zfighting3();
+    if (sig_addr3) {
+        if (patch_shader_trans_zfighting3a_fix.build(sig_addr3 +  0xE)
+         && patch_shader_trans_zfighting3b_fix.build(sig_addr3 + 0x34)
+         && patch_shader_trans_zfighting3c_fix.build(sig_addr3 +  0xE + 0x50)
+         && patch_shader_trans_zfighting3d_fix.build(sig_addr3 + 0x34 + 0x50)) {
```
Comment (Mimickal) [inline vulpes/fixes/shdr_trans_zfighting.cpp]: You know what I'm going to say about these constants.

File: vulpes/hooks/console.cpp [hunk 5]
```diff
@@ -8,48 +8,42 @@
...
+static Patch(console_in_hook_patch,
+    NULL, 6, JMP_PATCH, &console_hook__console_in);
 
 void init_console_input_hook() {
-    if(console_in_hook_patch.build()) {
-        console_hook__return_to_halo_con_in = console_in_hook_patch.address() + 23;
-        console_in_hook_patch.apply();
-    }
+    auto addr = sig_hook_console_input();
+    console_hook__return_to_halo_con_in = addr + 23;
```
Comment (Mimickal) [inline vulpes/hooks/console.cpp:26]: 23 is a magic number

File: vulpes/upgrades/map.cpp [hunk 6]
```diff
@@ -219,25 +190,41 @@ extern "C" {
...
-    if (patch_get_map_crc.build()) {
+    if (patch_read_map_file_header_replacement.build(sig_addr2))
+        patch_read_map_file_header_replacement.apply();
+    if (patch_startup_crc_calc_nop.build(sig_addr1))
+        patch_startup_crc_calc_nop.apply();
+    if (patch_get_map_crc.build(sig_map_crc_get_crc_from_table_hook())) {
         if (!map_upgrades_initialized) {
-            multiplayer_maps_list_ptr = *reinterpret_cast<uintptr_t**>(patch_get_map_crc.address()+2);
+            multiplayer_maps_list_ptr = *reinterpret_cast<uintptr_t**>(
+                patch_get_map_crc.address()+2);
```
Comment (Mimickal) [inline vulpes/upgrades/map.cpp]: Are 2 and 13 magic numbers?

File: vulpes/vulpes.cpp [hunk 7]
```diff
@@ -117,31 +117,44 @@ void init_network() {
...
 
+#include <vulpes/memory/signatures.hpp>
+
 void init_vulpes() {
     // Get safe search bounds for CodeSignature.
-    ImageSectionHeader* header = reinterpret_cast<ImageSectionHeader*>(sig_text_segment_data.address());
-    set_lowest_permitted_address(0x400000 + header->offset_to_segment);
-    set_highest_permitted_address(0x400000 + header->offset_to_segment + header->size_of_segment);
+    ImageSectionHeader* header = reinterpret_cast<ImageSectionHeader*>(
+        signature_text_segment_data.search(0x400000, 0x401000));
```
Comment (Mimickal) [inline vulpes/vulpes.cpp]: What are these values?

**Review by Mimickal — CHANGES_REQUESTED** [review]
Sorry this took so long

File: vulpes/memory/global.cpp (hunk 1 above)
Comment (ShellyHerself) [reply vulpes/memory/global.cpp:21]: This would get me an ISO C++ warning/error

**Review by Mimickal — APPROVED** [review]

**Pull request merged**